strapi/strapi
# fix(users): adjust response management to create user

### What does it do?

The problem is that the response from createUser service has two `data` keys in the response. This condition not fulfilled due to what I mentioned above: [Condition](https://github.com/strapi/strapi/blob/develop/packages/core/admin/admin/src/pages/Settings/pages/Users/components/NewUserForm.tsx#L114)

With this PR, the first `data` key of the response is taken to avoid failure in the validation process of the user's invitation token

### Why is it needed?
After new user is invited into Strapi with any user role there is error/warning. User is created on the background but registration token is not visible or email is send (Nodemailer). Tested on current project with Strapi 4.19.0, tested on fresh Strapi project with 4.19.0 - doesn't work, and tested on Strapi 4.15.5 - WORKS.

Unfortunately there is no console error log

![image](https://github.com/strapi/strapi/assets/73908004/a6bc6293-51e3-45d7-b285-bcc1086a737b)

Taken from: [4.19.0 - Warning on user invitation (an error occured) - doesn't generate activate link/email](https://github.com/strapi/strapi/issues/19352#issue-2104149695)

### How to test it?

* Exec `cd examples/getstarted && yarn develop`
* Go to create user process and create a user.
* At end of process we should view link of invitation
 
![image](https://github.com/strapi/strapi/assets/73908004/3836961f-2c8d-4bf9-8335-efafac9b75a3)



### Related issue(s)/PR(s)

[4.19.0 - Warning on user invitation (an error occured) - doesn't generate activate link/email](https://github.com/strapi/strapi/issues/19352#issue-2104149695)
